# Actividad 6 parte A

Yessica Abigail Santiago Hernandez 1990002 

GPO 43


In [26]:
#librerias basicas:

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

In [27]:
#librerias especializadas
from lifelines import CoxPHFitter, ExponentialFitter, WeibullFitter,LogNormalFitter, LogNormalAFTFitter
import seaborn as sns

In [28]:
#base de datos:
renuncias = pd.read_csv('https://raw.githubusercontent.com/jimmyzac/Estadistica-Aplicada-FCFM-UANL/main/bases_datos/desercion_laboral.csv')

In [29]:
#mostrar base de datos:
renuncias.head()

satisfaccion  evaluacion  proyectos  promedio_horas_mes  semestre  \
0           3.8         5.3          2                 157         3   
1           8.0         8.6          5                 262         6   
2           1.1         8.8          7                 272         4   
3           7.2         8.7          5                 223         5   
4           3.7         5.2          2                 159         3   

   accidente  renunciar  ascenso_5years departamento salario  
0          0          1               0       ventas    bajo  
1          0          1               0       ventas   medio  
2          0          1               0       ventas   medio  
3          0          1               0       ventas    bajo  
4          0          1               0       ventas    bajo

## Depuramiento de la base de datos 

1. Verificar que no haya missing values, en caso de haberlos decidir imputar o eliminar

In [30]:
#verificar la existencia de missin values 
renuncias.isnull().sum()

satisfaccion           0
evaluacion             0
proyectos              0
promedio_horas_mes     0
semestre               0
accidente              0
renunciar              0
ascenso_5years         0
departamento          26
salario                0
dtype: int64

In [31]:
#eliminar los datos:
renuncias = renuncias.dropna()

In [32]:
#verificar que se eliminaron esos datos
renuncias.isnull().sum()

satisfaccion          0
evaluacion            0
proyectos             0
promedio_horas_mes    0
semestre              0
accidente             0
renunciar             0
ascenso_5years        0
departamento          0
salario               0
dtype: int64

2. Verificar que no haya valores duplicados con dataframe.duplicated().sum() en caso de haber valores duplicados eliminarlos con dataframe=dataframe.drop_duplicates(keep='first').reset_index(drop=True)

In [33]:
#verificacion de valores duplicados
renuncias.duplicated().sum()

2982

In [34]:
#eliminar los valores duplicados 
renuncias =renuncias.drop_duplicates(keep='first').reset_index(drop=True)

In [35]:
#verificar que ya no existan esos valores duplicados 
renuncias.duplicated().sum()

0

3. Verificar  que  todas  las  variables  sean  numéricas,  si  no  son  numéricas hacerlas   numéricas,   en   el   caso   de   las   categóricas   convertirlas   en dummies y conservar n-1 dummies

In [36]:
#verifcar que sean variable numericas
renuncias.dtypes

satisfaccion          float64
evaluacion            float64
proyectos               int64
promedio_horas_mes      int64
semestre                int64
accidente               int64
renunciar               int64
ascenso_5years          int64
departamento           object
salario                object
dtype: object

In [37]:
# dummie departamento 
dum_dep = pd.get_dummies(renuncias['departamento'],drop_first=True,dtype=int) #el que se elimina es investigacion

In [38]:
dum_dep.head(2)

administración  contabilidad  desarrollo de producto  informática  \
0               0             0                       0            0   
1               0             0                       0            0   

   mantenimiento  mercadotecnia  recursos humanos  soporte  ventas  
0              0              0                 0        0       1  
1              0              0                 0        0       1

In [39]:
renuncias = pd.concat([renuncias,dum_dep],axis = 1)
renuncias = renuncias.drop('departamento',axis=1)

In [40]:
dum_sal = pd.get_dummies(renuncias['salario'],drop_first=True,dtype=int)

In [41]:
dum_sal.head()

bajo  medio
0     1      0
1     0      1
2     0      1
3     1      0
4     1      0

In [42]:
dum_sal = dum_sal[['bajo','medio']] 
renuncias = pd.concat([renuncias,dum_sal],axis = 1)
renuncias = renuncias.drop('salario',axis=1)

In [43]:
#resultado final de la base de datos 
renuncias.head()

satisfaccion  evaluacion  proyectos  promedio_horas_mes  semestre  \
0           3.8         5.3          2                 157         3   
1           8.0         8.6          5                 262         6   
2           1.1         8.8          7                 272         4   
3           7.2         8.7          5                 223         5   
4           3.7         5.2          2                 159         3   

   accidente  renunciar  ascenso_5years  administración  contabilidad  \
0          0          1               0               0             0   
1          0          1               0               0             0   
2          0          1               0               0             0   
3          0          1               0               0             0   
4          0          1               0               0             0   

   desarrollo de producto  informática  mantenimiento  mercadotecnia  \
0                       0            0              0              0   
1                       0            0              0              0   
2                       0            0              0              0   
3                       0            0              0              0   
4                       0            0              0              0   

   recursos humanos  soporte  ventas  bajo  medio  
0                 0        0       1     1      0  
1                 0        0       1     0      1  
2                 0        0       1     0      1  
3                 0        0       1     1      0  
4                 0        0       1     1      0

4. Calcular estadísticas descriptivas

In [44]:
renuncias.describe()

satisfaccion    evaluacion     proyectos  promedio_horas_mes  \
count  11991.000000  11991.000000  11991.000000        11991.000000   
mean       6.296581      7.166825      3.802852          200.473522   
std        2.410700      1.683426      1.163238           48.727813   
min        0.900000      3.600000      2.000000           96.000000   
25%        4.800000      5.700000      3.000000          157.000000   
50%        6.600000      7.200000      4.000000          200.000000   
75%        8.200000      8.600000      5.000000          243.000000   
max       10.000000     10.000000      7.000000          310.000000   

           semestre     accidente     renunciar  ascenso_5years  \
count  11991.000000  11991.000000  11991.000000    11991.000000   
mean       3.364857      0.154282      0.166041        0.016929   
std        1.330240      0.361234      0.372133        0.129012   
min        2.000000      0.000000      0.000000        0.000000   
25%        3.000000      0.000000      0.000000        0.000000   
50%        3.000000      0.000000      0.000000        0.000000   
75%        4.000000      0.000000      0.000000        0.000000   
max       10.000000      1.000000      1.000000        1.000000   

       administración  contabilidad  desarrollo de producto   informática  \
count    11991.000000  11991.000000            11991.000000  11991.000000   
mean         0.036361      0.051789                0.057210      0.081394   
std          0.187194      0.221610                0.232252      0.273451   
min          0.000000      0.000000                0.000000      0.000000   
25%          0.000000      0.000000                0.000000      0.000000   
50%          0.000000      0.000000                0.000000      0.000000   
75%          0.000000      0.000000                0.000000      0.000000   
max          1.000000      1.000000                1.000000      1.000000   

       mantenimiento  mercadotecnia  recursos humanos       soporte  \
count    11991.00000   11991.000000      11991.000000  11991.000000   
mean         0.18714       0.056125          0.050121      0.151864   
std          0.39004       0.230173          0.218204      0.358904   
min          0.00000       0.000000          0.000000      0.000000   
25%          0.00000       0.000000          0.000000      0.000000   
50%          0.00000       0.000000          0.000000      0.000000   
75%          0.00000       0.000000          0.000000      0.000000   
max          1.00000       1.000000          1.000000      1.000000   

             ventas          bajo         medio  
count  11991.000000  11991.000000  11991.000000  
mean       0.270119      0.478692      0.438746  
std        0.444040      0.499567      0.496254  
min        0.000000      0.000000      0.000000  
25%        0.000000      0.000000      0.000000  
50%        0.000000      0.000000      0.000000  
75%        1.000000      1.000000      1.000000  
max        1.000000      1.000000      1.000000

## Estimar el modelode Cox e interpretar cada uno de los coeficientes

In [45]:
cph = CoxPHFitter().fit(renuncias, 'semestre', 'renunciar')
cph.print_summary()

<lifelines.CoxPHFitter: fitted with 11991 total observations, 10000 right-censored observations>
             duration col = 'semestre'
                event col = 'renunciar'
      baseline estimation = breslow
   number of observations = 11991
number of events observed = 1991
   partial log-likelihood = -15884.05
         time fit was run = 2023-09-27 03:14:39 UTC

---
                         coef  exp(coef)   se(coef)   coef lower 95%   coef upper 95%  exp(coef) lower 95%  exp(coef) upper 95%
covariate                                                                                                                      
satisfaccion            -0.24       0.78       0.01            -0.26            -0.22                 0.77                 0.80
evaluacion              -0.01       0.99       0.01            -0.04             0.02                 0.96                 1.02
proyectos               -0.29       0.75       0.02            -0.33            -0.25                 0.72                 0.78
promedio_horas_mes       0.00       1.00       0.00             0.00             0.00                 1.00                 1.00
accidente               -1.25       0.29       0.10            -1.44            -1.05                 0.24                 0.35
ascenso_5years          -1.38       0.25       0.36            -2.08            -0.68                 0.13                 0.51
administración           0.22       1.25       0.18            -0.13             0.57                 0.88                 1.76
contabilidad             0.24       1.27       0.14            -0.04             0.52                 0.96                 1.69
desarrollo de producto   0.36       1.43       0.14             0.08             0.64                 1.08                 1.90
informática              0.29       1.34       0.13             0.03             0.56                 1.03                 1.74
mantenimiento            0.43       1.54       0.12             0.20             0.67                 1.22                 1.95
mercadotecnia            0.34       1.40       0.14             0.06             0.62                 1.06                 1.86
recursos humanos         0.43       1.54       0.14             0.15             0.72                 1.16                 2.05
soporte                  0.45       1.57       0.12             0.21             0.69                 1.23                 2.00
ventas                   0.35       1.42       0.12             0.12             0.58                 1.13                 1.78
bajo                     1.54       4.66       0.15             1.25             1.83                 3.48                 6.23
medio                    1.15       3.15       0.15             0.85             1.44                 2.35                 4.22

                         cmp to      z      p   -log2(p)
covariate                                               
satisfaccion               0.00 -26.89 <0.005     526.58
evaluacion                 0.00  -0.76   0.44       1.17
proyectos                  0.00 -14.75 <0.005     161.15
promedio_horas_mes         0.00   3.88 <0.005      13.25
accidente                  0.00 -12.41 <0.005     115.04
ascenso_5years             0.00  -3.88 <0.005      13.21
administración             0.00   1.24   0.21       2.22
contabilidad               0.00   1.66   0.10       3.36
desarrollo de producto     0.00   2.49   0.01       6.29
informática                0.00   2.17   0.03       5.06
mantenimiento              0.00   3.60 <0.005      11.64
mercadotecnia              0.00   2.35   0.02       5.75
recursos humanos           0.00   3.02 <0.005       8.64
soporte                    0.00   3.68 <0.005      12.08
ventas                     0.00   3.01 <0.005       8.56
bajo                       0.00  10.36 <0.005      81.15
medio                      0.00   7.67 <0.005      45.68
---
Concordance = 0.85
Partial AIC = 31802.11
log-likelihood ratio test = 1360.94 on 17 df
-log2(p) of ll-rati

**satisfaccion:**
p-value: 0.005<0.05. Rechazamos H0 lo que significa que el nivel se satisfaccion en un empleado si afecta el riesgo de que una persona renuncie

Por cada nivel de satisfaccion de un empleado en su trabajo se reduce el riesgo de que renuncie en un 22% respecto nivel de satisfaccion

**evaluacion:**
p-value: 0.44>0.05. No rechazamos H0 lo que significa que el nivel de evaluacion en su forma de hacer el trabajo (que tan bien lo realiza) no afecta el riesgo de que una persona renuncie.

**proyectos:**
p-value: <0.005>0.05  Rechazamos H0 lo que significa que el número de proyectos asignados al empleado si afecta el riesgo de que una persona renuncie

Por cada número de proyectos asignados al empleado en su trabajo se reduce el riesgo de que renuncie en un 25% 

**promedio horas mes:**
p-value: <0.005>0.05  Rechazamos H0 lo que significa que el promedio de horas mensuales trabajadas si afecta el riesgo de que una persona renuncie

Por cada hora de que un empleado trabja al mes se reduce el riesgo de que renuncie en un .18%

**accidente:**
p-value: <0.005>0.05  Rechazamos H0 lo que significa que haber tenido un accidente dentro de la empresa si afecta el riesgo de que una persona renuncie

El haber tenido un accidente en el trabajo reduce el riesgo de que renuncie en un 71% respecto a si no ha tenido un accidente 

**ascenso 5 años:**
p-value: <0.005>0.05  Rechazamos H0 lo que significa que si el empleado ha sido ascendido durante los últimos 5 años si afecta el riesgo de que una persona renuncie

Si el empleado ha sido ascendido durante los últimos 5 años reduce el riesgo de que renuncie en un 75% respecto a si no ha sido ascendido durante los últimos 5 años

**administracion:**
p-value: 0.44>0.05. No rechazamos H0 lo que significa que el que el empleado trabaje en el departamento de administración no afecta en el riesgo de que una persona renuncie en comparacion a ser parte del departamento de investigacion

**contabilidad:**
p-value: 0.44>0.05. No rechazamos H0 lo que significa que el que el empleado trabaje en el departamento de contabilidad no afecta en el riesgo de que una persona renuncie en comparacion a ser parte del departamento de investigacion

**desarrollo de producto:**
p-value: 0.01<0.05 Rechazamos H0 lo que significa que estar en el departamento de desarrollo de producto si afecta el riesgo de que una persona renuncie 

Estar en el departamento de desarrollo de producto aumenta el riesgo de que renuncie en un 43% en comparacion a ser parte del departamento de investigacion

**informatica:**
p-value: 0.03<0.05 Rechazamos H0 lo que significa que estar en el departamento de informatica de producto si afecta el riesgo de que una persona renuncie 

Estar en el departamento de informatica aumenta el riesgo de que renuncie en un 34% en comparacion a ser parte del departamento de investigacion

**mantenimiento:**
p-value: <0.005<0.05 Rechazamos H0 lo que significa que estar en el departamento de mantenimiento si afecta el riesgo de que una persona renuncie 

Estar en el departamento de mantenimiento aumenta el riesgo de que renuncie en un 54% en comparacion a ser parte del departamento de investigacion

**mercadotecnia:**
p-value: 0.02<0.05 Rechazamos H0 lo que significa que estar en el departamento de mercadotecnia si afecta el riesgo de que una persona renuncie 

Estar en el departamento de mercadotecnia aumenta el riesgo de que renuncie en un 40% en comparacion a ser parte del departamento de investigacion

**recursos humanos:**
p-value: <0.005<0.05 Rechazamos H0 lo que significa que estar en el departamento de recursos humanos si afecta el riesgo de que una persona renuncie 

Estar en el departamento de recursos humanos aumenta el riesgo de que renuncie en un 54% en comparacion a ser parte del departamento de investigacion

**soporte:**
p-value: <0.05<0.05 Rechazamos H0 lo que significa que estar en el departamento de soporte si afecta el riesgo de que una persona renuncie 

Estar en el departamento de soporte aumenta el riesgo de que renuncie en un 40% en comparacion a ser parte del departamento de investigacion

**ventas:**
p-value: <0.05<0.05 Rechazamos H0 lo que significa que estar en el departamento de ventas si afecta el riesgo de que una persona renuncie 

Estar en el departamento de soporte aumenta el riesgo de que renuncie en un 42% en comparacion a ser parte del departamento de investigacion

**bajo:**
p-value: <0.05<0.05 Rechazamos H0 lo que significa que tener un salario bajo si afecta el riesgo de que una persona renuncie 

Tener un salario bajo aumenta el riesgo de que renuncie en un 366% en comparacion a tener un salario alto

**medio:**
p-value: <0.005<0.05 Rechazamos H0 lo que significa que tener un salario medio si afecta el riesgo de que una persona renuncie 

Tener un salario medio aumenta el riesgo de que renuncie en un 215% en comparacion a tener un salario alto

## Con base  en la interpretación de los coeficientes,¿quérecomendación podría hacerle  a  la  empresa  para  que  disminuya  la  cantidad  de  renuncias  de  los empleados?

1. buscar que los empleados se sientan mas satisfechos con sus trabajos
2. aumentar los salarios de los empleados, este punto es de suma importancia puesto que tiene altos porcentajes de afectacion 
3. Buscar la seguridad de los empleados es vital, puesto que tambien tiene un gran impacto en el riesgo de que un empleado renuncie 
4. Buscar hacer mas atractivos los departamentos que si tienen una afectacion como informatica, mantenimiento, mercadotecnia, recursos humanos, etc.

## Pruebe si el modelo de Cox cumple el supuesto de riesgosproporcionales, de no hacerlo estime el modelo AFT adecuado

In [46]:
cph.check_assumptions(renuncias,p_value_threshold=0.05)
#Ho se viola el supuesto
#Ha no se viola el supuesto

The ``p_value_threshold`` is set at 0.05. Even under the null hypothesis of no violations, some
covariates will be below the threshold by chance. This is compounded when there are many covariates.
Similarly, when there are lots of observations, even minor deviances from the proportional hazard
assumption will be flagged.

With that in mind, it's best to use a combination of statistical tests and visual tests to determine
the most serious violations. Produce visual plots using ``check_assumptions(..., show_plots=True)``
and looking for non-constant lines. See link [A] below for a full example.



<lifelines.StatisticalResult: proportional_hazard_test>
 null_distribution = chi squared
degrees_of_freedom = 1
             model = <lifelines.CoxPHFitter: fitted with 11991 total observations, 10000 right-censored observations>
         test_name = proportional_hazard_test

---
                             test_statistic      p  -log2(p)
accidente              km              0.03   0.87      0.21
                       rank            1.14   0.29      1.81
administración         km              5.08   0.02      5.37
                       rank            5.74   0.02      5.92
ascenso_5years         km              3.66   0.06      4.17
                       rank            6.71   0.01      6.71
bajo                   km              0.37   0.54      0.88
                       rank            0.00   0.96      0.06
contabilidad           km              0.09   0.77      0.39
                       rank            1.14   0.29      1.81
desarrollo de producto km              1.03   0.31      1.69
                       rank            0.85   0.36      1.49
evaluacion             km            443.05 <0.005    324.32
                       rank          427.15 <0.005    312.82
informática            km              0.48   0.49      1.03
                       rank            0.53   0.47      1.10
mantenimiento          km              0.47   0.49      1.02
                       rank            2.26   0.13      2.91
medio                  km              0.16   0.69      0.53
                       rank            0.10   0.75      0.42
mercadotecnia          km              2.66   0.10      3.28
                       rank            6.33   0.01      6.39
promedio_horas_mes     km            270.84 <0.005    199.74
                       rank          307.41 <0.005    226.21
proyectos              km            574.13 <0.005    419.05
                       rank          684.93 <0.005    499.11
recursos humanos       km              0.68   0.41      1.29
                       rank            2.86   0.09      3.46
satisfaccion           km           1257.74 <0.005    912.74
                       rank          845.24 <0.005    614.90
soporte                km              0.00   0.98      0.03
                       rank            0.89   0.35      1.53
ventas                 km              1.04   0.31      1.70
                       rank            4.36   0.04      4.76



1. Variable 'satisfaccion' failed the non-proportional test: p-value is <5e-05.

   Advice 1: the functional form of the variable 'satisfaccion' might be incorrect. That is, there
may be non-linear terms missing. The proportional hazard test used is very sensitive to incorrect
functional forms. See documentation in link [D] below on how to specify a functional form.

   Advice 2: try binning the variable 'satisfaccion' using pd.cut, and then specify it in
`strata=['satisfaccion', ...]` in the call in `.fit`. See documentation in link [B] below.

   Advice 3: try adding an interaction term with your time variable. See documentation in link [C]
below.


2. Variable 'evaluacion' failed the non-proportional test: p-value is <5e-05.

   Advice 1: the functional form of the variable 'evaluacion' might be incorrect. That is, there may
be non-linear terms missing. The proportional hazard test used is very sensitive to incorrect
functional forms. See documentation in link [D] below on how to 

[]

In [47]:
mexpo = ExponentialFitter().fit(renuncias['semestre'],renuncias['renunciar'])
mweibull = WeibullFitter().fit(renuncias['semestre'],renuncias['renunciar'])
mlogn = LogNormalFitter().fit(renuncias['semestre'],renuncias['renunciar'])

In [48]:
mexpo.AIC_.round(2), mweibull.AIC_.round(2), mlogn.AIC_.round(2)

(15965.46, 12511.51, 11587.92)

In [49]:
#estimamos un log
lognorAFT = LogNormalAFTFitter().fit(renuncias, 'semestre', 'renunciar')
lognorAFT.print_summary()

<lifelines.LogNormalAFTFitter: fitted with 11991 total observations, 10000 right-censored observations>
             duration col = 'semestre'
                event col = 'renunciar'
   number of observations = 11991
number of events observed = 1991
           log-likelihood = -4876.57
         time fit was run = 2023-09-27 03:14:42 UTC

---
                                coef  exp(coef)   se(coef)   coef lower 95%   coef upper 95%  exp(coef) lower 95%  exp(coef) upper 95%
param  covariate                                                                                                                      
mu_    accidente                0.26       1.30       0.02             0.22             0.30                 1.25                 1.35
       administración           0.05       1.05       0.04            -0.03             0.13                 0.97                 1.13
       ascenso_5years           0.37       1.44       0.06             0.24             0.49                 1.27                 1.63
       bajo                    -0.37       0.69       0.03            -0.43            -0.31                 0.65                 0.73
       contabilidad            -0.04       0.96       0.03            -0.11             0.02                 0.90                 1.02
       desarrollo de producto  -0.05       0.95       0.03            -0.12             0.01                 0.89                 1.01
       evaluacion               0.01       1.01       0.00            -0.00             0.01                 1.00                 1.01
       informática             -0.05       0.95       0.03            -0.11             0.01                 0.90                 1.02
       mantenimiento           -0.08       0.93       0.03            -0.13            -0.02                 0.88                 0.98
       medio                   -0.29       0.75       0.03            -0.34            -0.23                 0.71                 0.80
       mercadotecnia           -0.04       0.97       0.03            -0.10             0.03                 0.90                 1.03
       promedio_horas_mes      -0.00       1.00       0.00            -0.00            -0.00                 1.00                 1.00
       proyectos                0.10       1.10       0.01             0.09             0.11                 1.09                 1.11
       recursos humanos        -0.09       0.92       0.03            -0.15            -0.02                 0.86                 0.98
       satisfaccion             0.07       1.07       0.00             0.06             0.07                 1.06                 1.07
       soporte                 -0.08       0.92       0.03            -0.14            -0.03                 0.87                 0.97
       ventas                  -0.06       0.94       0.03            -0.11            -0.00                 0.90                 1.00
       Intercept                1.34       3.83       0.04             1.25             1.43                 3.50                 4.18
sigma_ Intercept               -1.11       0.33       0.02            -1.14            -1.08                 0.32                 0.34

                                cmp to      z      p   -log2(p)
param  covariate                                               
mu_    accidente                  0.00  12.98 <0.005     125.59
       administración             0.00   1.21   0.23       2.14
       ascenso_5years             0.00   5.80 <0.005      27.19
       bajo                       0.00 -12.49 <0.005     116.53
       contabilidad               0.00  -1.27   0.20       2.30
       desarrollo de producto     0.00  -1.54   0.12       3.01
       evaluacion                 0.00   1.46   0.14       2.80
       informática                0.00  -1.48   0.14       2.85
       mantenimiento              0.00  -2.79   0.01       7.58
       medio                      0.00  -9.59 <0.005      69.93
       mercadotecnia              0.00  -1.05   0.29

## Interprete los coeficientes del modelo AFT

**accidente**
p-value: <0.005<0.05 rechazamos Ho.  El haber tenido un accidente si tiene efecto sobre el rego de renunciar 
 
 Haber sufrido un accidente dentro de la empresa aumenta la mediana del tiempo de supervivencia 30% con respecto a no haber sufrido algun accidente.

**administración**
p-value: 0.23>0.05 no rechazamos Ho. Estar en el departamento de administracion no tiene efecto, pertenecer al departamento de administración no tiene efecto sobre la mediana del tiempo de supervivencia.

**ascenso 5 a;os**
p-value: <0.005<0.05 rechazamos Ho. el haber tenido un ascenso en los ultimos 5 a;os si tiene efecto sobre el riesgo de que un empleado renuncie 

Haber sido ascendido durante los ultimos 5 años aumenta la mediana del tiempo de supervivencia 44% con respecto a no haber sido ascendido.

**salario bajo**
p-value<0.005<0.05, rechazamos Ho.Recibir un salario bajo disminuye la mediana del tiempo de supervivencia 31% con respecto a recibir un salario alto.

**contabilidad**
p-value: 0.20>0.05 no rechazamos Ho. Estar en el departamento de contabilidad  no tiene efecto sobre la mediana del tiempo de supervivencia.

**desarrollo de producto**
p-value: 0.12>0.05, no rechazamos Ho. Estar en el departamento de desarrollo de producto no tiene efecto sobre la mediana del tiempo de supervivencia.

**evaluacion**
p-value: 0.14>0.05, no rechazamos Ho. El que tan bien realiza su trabajo el empleado no tiene efecto sobre la mediana del tiempo de supervivencia.

**informática**
p-value: 0.14>0.05, no rechazamos Ho. Estar en el departamento de informática no tiene efecto sobre la mediana del tiempo de supervivencia.

**mantenimiento**
p-value: 0.01<0.05, rechazamos Ho. Estar en el departamento de mantenimiento disminuye la mediana del tiempo de supervivencia 7% con respecto a pertenecer al departamento de investigación.

**salario medio**
p-value<0.005<0.05, rechazamos Ho. Recibir un salario medio disminuye la mediana del tiempo de supervivencia 25% con respecto a recibir un salario alto.

**mercadotecnia**
p-value: 0.29>0.05, no rechazamos Ho. Estar en eldepartamento de mercadotecnia no tiene efecto sobre la mediana del tiempo de supervivencia.

**promedio horas mes**
p-value: 0.02<0.05, rechazamos Ho. Cada hora que se trabaja al mes disminuyen la mediana del tiempo de supervivencia 0.0309%.

**proyectos**
p-value: <0.005<0.05, rechazamos Ho. Cada proyecto que se le asignó al trabajador aumenta la mediana del tiempo de supervivencia 10%.

**recursos humanos**
p-value: 0.01<0.05, rechazamos Ho. Estar en el departamento de recursos humanos disminuye la mediana del tiempo de supervivencia 8% con respecto a pertenecer al departamento de investigación.

**satisfaccion**
p-value<0.005<0.05=alfa, rechazamos Ho. El nivel de satisfaccion de los empleados por su trabajo aumenta la mediana del tiempo de supervivencia 7% con respecto al punto anterior.

**soporte**
p-value: <0.005<0.05, rechazamos Ho. Estar en el departamento de soporte disminuye la mediana del tiempo de supervivencia 8% con respecto a pertenecer al departamento de investigación.

**ventas**
p-value: 0.03<0.05, rechazamos Ho. Estar en el departamento de ventas disminuye la mediana del tiempo de supervivencia 6% con respecto a pertenecer al departamento de investigación.
